corehq/motech/repeater_helpers.py

```diff
@@ -24,22 +24,28 @@ def get_relevant_case_updates_from_form_json(domain, form_json, case_types, extr
     case_blocks = extract_case_blocks(form_json)
     cases = CaseAccessors(domain).get_cases(
         [case_block['@case_id'] for case_block in case_blocks], ordered=True)
-    for case, case_block in zip(cases, case_blocks):
-        assert case_block['@case_id'] == case.case_id
+
+    db_case_dict = {case.case_id: case for case in cases}
+
+    for case_block in case_blocks:
+        case = db_case_dict[case_block['@case_id']]
+
         if not case_types or case.type in case_types:
             case_create = case_block.get('create') or {}
             case_update = case_block.get('update') or {}
-            result.append(CaseTriggerInfo(
-                domain=domain,
-                case_id=case_block['@case_id'],
-                type=case.type,
-                name=case.name,
-                owner_id=case.owner_id,
-                modified_by=case.modified_by,
-                updates={**case_create, **case_update},
-                created='create' in case_block,
-                closed='close' in case_block,
-                extra_fields={field: case.get_case_property(field) for field in extra_fields},
-                form_question_values=form_question_values or {},
-            ))
+
+            if case_create or case_update:
```

Comment:
Ah, fair enough. I added the `if` clause because it did not make sense to consider the iteration if it does not create or update anything, but I missed the fact that cases can/will be closed also.